### PR TITLE
Add missing docstrings for Node methods

### LINKED ISSUE #235: Missing docstrings
It was noted that `Node.config()` is not included in API docs. This particular method does not have a docstring. Per [interrogate](https://interrogate.readthedocs.io/en/latest/), a third of fablib do not have docstrings:

```console
$ interrogate -v fabrictestbed_extensions/fablib/ 
----------------------------------- Summary ------------------------------------
| Name                     |      Total |      Miss |      Cover |      Cover% |
|--------------------------|------------|-----------|------------|-------------|
| __init__.py              |          1 |         1 |          0 |          0% |
| component.py             |         40 |        12 |         28 |         70% |
| fablib.py                |        115 |        29 |         86 |         75% |
| facility_port.py         |         18 |        12 |          6 |         33% |
| interface.py             |         63 |        27 |         36 |         57% |
| network_service.py       |         67 |        31 |         36 |         54% |
| node.py                  |        124 |        40 |         84 |         68% |
| resources.py             |         43 |         8 |         35 |         81% |
| slice.py                 |         84 |        20 |         64 |         76% |
|--------------------------|------------|-----------|------------|-------------|
| TOTAL                    |        555 |       180 |        375 |       67.6% |
---------------- RESULT: FAILED (minimum: 80.0%, actual: 67.6%) ----------------
```

### PR DESCRIPTION
**Pull request by sajith**
Partly resolves #235.

It is not immediately apparent which methods really _deserve_ docstrings, so I went though the notebook examples, and documented the methods that appear to be public.  The rest should be either marked as private, or documented, but this will require some consideration.

### CODE REVIEW

**Comment by sajith** [comment]
Note that there still many methods with missing docstrings.  I am not sure if some of them should be private (such as `Node.run_post_boot_task()`) or semi-private (`Node.config()` and `Node.delete()`, which are rarely called directly, if I understand correctly.)

Here's a list of remaining missing docstrings:

```
|     Node.get_fablib_manager (L100)                             |      MISSED |
|     Node.get_pretty_name_dict (L209)                           |      MISSED |
|     Node.generate_template_context (L281)                      |      MISSED |
|     Node.get_template_context (L291)                           |      MISSED |
|     Node.render_template (L294)                                |      MISSED |
|     Node.delete (L302)                                         |      MISSED |
|     Node.get_networks (L631)                                   |      MISSED |
|     Node.get_paramiko_key (L1186)                              |      MISSED |
|     Node.get_fim (L2728)                                       |      MISSED |
|     Node.set_user_data (L2731)                                 |      MISSED |
|     Node.get_user_data (L2736)                                 |      MISSED |
|     Node.delete (L2742)                                        |      MISSED |
|     Node.init_fablib_data (L2748)                              |      MISSED |
|     Node.get_fablib_data (L2757)                               |      MISSED |
|     Node.set_fablib_data (L2763)                               |      MISSED |
|     Node.get_routes (L2879)                                    |      MISSED |
|     Node.config_routes (L2885)                                 |      MISSED |
|     Node.run_post_boot_tasks (L2910)                           |      MISSED |
|     Node.run_post_update_commands (L2944)                      |      MISSED |
|     Node.is_instantiated (L2956)                               |      MISSED |
|     Node.set_instantiated (L2975)                              |      MISSED |
|     Node.run_update_commands (L2980)                           |      MISSED |
|     Node.set_run_update_commands (L2987)                       |      MISSED |
|     Node.config (L2992)                                        |      MISSED |
|     Node.add_fabnet (L3008)                                    |      MISSED |
```

Some are buggy, such as `Node.list_networks()` (#239).

File: fabrictestbed_extensions/fablib/node.py [hunk 1]
```diff
@@ -2934,6 +2990,25 @@ def set_run_update_commands(self, run_update_commands: bool = True):
         self.set_fablib_data(fablib_data)
 
     def config(self, log_dir="."):
+        """
```
Comment (kthare10) [inline fabrictestbed_extensions/fablib/node.py:2993]: Can we also mention a NOTE here that the user can use this to re-apply config post os_reboot as well?

Comment (sajith) [reply fabrictestbed_extensions/fablib/node.py:2993]: Done. Please take a look: https://fabric-fablib--237.org.readthedocs.build/en/237/node.html#fabrictestbed_extensions.fablib.node.Node.config

Comment (kthare10) [reply fabrictestbed_extensions/fablib/node.py:2993]: Looks good!

**Review by kthare10 — APPROVED** [review]
Looks good!

**Pull request merged**